R/make_mixture_from_samples.R

```diff
@@ -81,13 +84,13 @@ mixture_from_samples <- function(data,
     # get an integer value of how many samples to draw from each model
     # note: instead of a rounding with preserved sum we might want to add
     # some randomness in where we round up so that there is no systematic bias
-    num_draws <- round_with_preserved_sum(S*weights)
+    num_draws <- round_with_preserved_sum(S * weights)
     
     # draw from individual models
     # note: maybe switch to purrr here
     mixture_vector <- sapply(seq_along(num_draws), 
                              FUN = function(i) {
-                               sample(x = dat[, get(..models[i])], 
+                               sample(x = dat[, models[i], with = FALSE], 
```

Comment:
this returns a data.table instead of a vector (which sample needs). Could do dat[, models[i], with = FALSE][[1]]? not sure that is better